blast/formatter/server/internal/server/run-report-ep.go

```diff
@@ -58,107 +56,54 @@ func runReport(job *api.JobPayload, log *logrus.Entry) midl.Response {
 	// If the directory does not exist, create it and run the report.
 	if err = runIfNeeded(cmd, outputDir, log); err != nil {
 		log.Error(err)
+		if e := xfiles.WriteFailedFlag(outputDir); e != nil {
+			log.Error(e)
+		}
 		return New500Error(err.Error())
 	}
 
 	log.Debug("Zipping command output.")
 	err = zipDir(outputDir)
 	if err != nil {
 		log.Error(err.Error())
+		if e := xfiles.WriteFailedFlag(outputDir); e != nil {
+			log.Error(e)
+		}
 		return New500Error(err.Error())
 	}
 
 	log.Debug("Writing out meta.json")
 
 	if err = WriteMeta(outputDir); err != nil {
 		log.Error(err.Error())
-		return New500Error(err.Error())
-	}
-
-	return New200Response("Job completed successfully")
-}
-
-func zipDir(path string) (err error) {
```

Comment:
The zip functions were moved to their own file to trim down the bloated controller file.